### PR TITLE
Use fcntl and FD_CLOEXEC if O_CLOEXEC is undefined

### PR DESCRIPTION
**Pull request by acesso**
Same reported at #87 but at src/shrpx_config.cc src/instead of util.cc

### CODE REVIEW

**Comment by tatsuhiro-t** [comment]
Thank you for PR!  Could you check my comment?

**Comment by acesso** [comment]
fixed and tested, compiled like a charm

File: src/shrpx_config.cc [hunk 1]
```diff
@@ -242,8 +242,18 @@ read_tls_ticket_key_file(const std::vector<std::string> &files) {
...
+  #if defined O_CLOEXEC
+    auto fd = open(filename, O_WRONLY | O_CLOEXEC | O_CREAT | O_TRUNC,
+                     S_IRUSR | S_IWUSR);
+ #else
+    auto fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC,
+                   S_IRUSR | S_IWUSR);
+ 
+    // We get race condition if execve is called at the same time.
+    if(fd != -1) {
+        int make_socket_closeonexec_(fd);
```
Comment (tatsuhiro-t) [inline src/shrpx_config.cc]: This is not correct to me.
I proposed specifically `util::make_socket_closeonexec(fd);`

**Comment by tatsuhiro-t** [comment]
Merged via 7ea8037ee1a274cb450b0453736cc793c3022407
Thank you!

**Pull request closed without merging**